packages/modules/cockroachdb/src/cockroachdb-container.ts

```diff
@@ -0,0 +1,79 @@
+import { AbstractStartedContainer, GenericContainer, StartedTestContainer, Wait } from "testcontainers";
+
+const COCKROACH_PORT = 26257;
+const COCKROACH_HTTP_PORT = 26258;
+
+export class CockroachDbContainer extends GenericContainer {
+  private database = "test";
+  private username = "test";
+
+  constructor(image = "cockroachdb/cockroach:v24.3.5") {
+    super(image);
+    this.withExposedPorts(COCKROACH_PORT, COCKROACH_HTTP_PORT)
+      .withCommand(["start-single-node", "--insecure", "--http-addr=0.0.0.0:" + COCKROACH_HTTP_PORT])
+      .withHealthCheck({
+        test: ["CMD-SHELL", "curl -f http://localhost:" + COCKROACH_HTTP_PORT + " || exit 1"],
+        interval: 1000,
+        timeout: 3000,
+        retries: 5,
+        startPeriod: 1000,
+      })
+      .withWaitStrategy(Wait.forHealthCheck());
+  }
+
+  public withDatabase(database: string): this {
+    this.database = database;
+    return this;
+  }
+
+  public withUsername(username: string): this {
+    this.username = username;
+    return this;
+  }
+
+  public override async start(): Promise<StartedCockroachDbContainer> {
+    this.withEnvironment({
+      COCKROACH_DATABASE: this.database,
+      COCKROACH_USER: this.username,
+    });
+    return new StartedCockroachDbContainer(await super.start(), this.database, this.username);
+  }
+}
+
+export class StartedCockroachDbContainer extends AbstractStartedContainer {
+  private port: number;
+
+  constructor(
+    startedTestContainer: StartedTestContainer,
+    private readonly database: string,
+    private readonly username: string
+  ) {
+    super(startedTestContainer);
+    this.port = startedTestContainer.getMappedPort(COCKROACH_PORT);
+  }
+
+  public getPort(): number {
+    this.port = this.startedTestContainer.getMappedPort(COCKROACH_PORT);
```

Comment:
Could just `return this.startedTestContainer.getMappedPort(COCKROACH_PORT);` and remove `this.port`

Reply:
haha good point 🤦